patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -3765,6 +3765,23 @@
  					int num178 = Projectile.NewProjectile(projectileSource_Item_WithPotentialAmmo, pointPoisition.X, pointPoisition.Y, num2, num3, projToShoot, Damage, KnockBack, i);
  					if (sItem.type == 726)
  						Main.projectile[num178].magic = true;
+@@ -34203,8 +_,14 @@
+ 				else if (num2 >= 427)
+ 					num = num2 - 426;
+ 
++				int num3 = 0;
++				if (num >= 22) {
++					sItem.ModItem.TorchVFX(out R, out G, out B, out num3);
++				}
```

Comment:
What does num stand for here? And concider a safety null check via null coalesc
```suggestion
+				if (num >= 22) {
+					sItem.ModItem?.TorchVFX(out R, out G, out B, out num3);
+				}
```

Reply:
Yea now that I am looking into it I don't really know why I did that so I have changed it to a safer null check on `ModItem` which should do the job just as well without the potential issues of how I was doing it.